sycl/include/sycl/ext/oneapi/experimental/bfloat16.hpp

```diff
@@ -63,10 +63,13 @@ class bfloat16 {
     return __spirv_ConvertBF16ToFINTEL(a);
 #endif
 #else
-    // Shift temporary variable to silence the warning
-    uint32_t bits = a;
-    bits <<= 16;
-    return static_cast<float>(bits);
+    union {
+      uint32_t bits;
+      float res;
+    } val;
+    val.bits = a;
+    val.bits <<= 16;
+    return val.res;
```

Comment:
We seem to have `sycl::bit_cast`, can we use it here? If not, then a comment is necessary to explain why.

Reply:
good point, will fix